### PR TITLE
Show distance from GPS lock position to current cursor 

### PR DESCRIPTION
**Pull request by nyalldawson**
When a GPS device is connected, whenever the user moves the cursor over the canvas a live status bar message displays the distance and bearing from the cursor to the GPS fix position.

Project distance and bearing settings are respected in this display.

Peek 2020-01-14 16-06

### CODE REVIEW

File: src/app/gps/qgsgpsinformationwidget.cpp [hunk 1]
```diff
@@ -1461,6 +1487,38 @@ void QgsGpsInformationWidget::timestampFormatChanged( int )
...
+
+  }
+}
+
+void QgsGpsInformationWidget::updateGpsDistanceStatusMessage()
+{
+  if ( !mNmea )
+    return;
+
+  QgsDistanceArea da;
```
Comment (nirvn) [inline src/app/gps/qgsgpsinformationwidget.cpp]: Out of curiosity, is that QgsDistanceArea expensive to create (setEllipsoid, setSourceCrs, etc.)? If the answer is yes, should it instead be a member of the QgsGpsInformationWidget class, and update the ellipsoid / source CRS through project signals?

Comment (nyalldawson) [reply src/app/gps/qgsgpsinformationwidget.cpp]: Good point, I'll do that

Comment (nyalldawson) [reply src/app/gps/qgsgpsinformationwidget.cpp]: Done

**Comment by nirvn** [comment]
@nyalldawson , merge away! :)

**Pull request merged**